client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,10 +194,8 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
-
-			return domainManagementList( domainName );
+		} else if ( cart.create_new_blog && ! cartItems.hasPlan( cart ) ) {
+			return domainManagementList( map( values( receipt.purchases )[ 0 ], 'meta' )[ 0 ] );
```

Comment:
Yes. Idea is to use recipe, because there's a racing condition here for cartItems. :) That's why Stephane saw the thank you page.